crates/agglayer-certificate-orchestrator/src/tests/certifier_results.rs

```diff
@@ -0,0 +1,882 @@
+use std::time::Duration;
+
+use agglayer_storage::stores::PendingCertificateReader;
+use agglayer_storage::stores::PendingCertificateWriter;
+use agglayer_storage::stores::StateReader;
+use agglayer_storage::stores::StateWriter;
+use agglayer_types::Certificate;
+use agglayer_types::CertificateHeader;
+use agglayer_types::Proof;
+use pessimistic_proof::LocalNetworkState;
+use pessimistic_proof::Signature;
+use pessimistic_proof::U256;
+use rstest::rstest;
+use tokio::time::sleep;
+
+use crate::tests::create_orchestrator;
+
+#[rstest]
+#[tokio::test]
+#[timeout(Duration::from_millis(100))]
+async fn certifier_results_for_unknown_network_with_height_zero() {
+    let (_data_sender, mut receiver, mut orchestrator) = create_orchestrator::default();
+
+    let certificate = Certificate {
+        network_id: 1.into(),
+        height: 0,
+        prev_local_exit_root: [0; 32],
+        new_local_exit_root: [0; 32],
+        bridge_exits: Vec::new(),
+        imported_bridge_exits: Vec::new(),
+        signature: Signature {
+            r: U256::ZERO,
+            s: U256::ZERO,
+            odd_y_parity: false,
+        },
+    };
+    let certificate2 = Certificate {
+        network_id: 1.into(),
+        height: 1,
+        prev_local_exit_root: [0; 32],
+        new_local_exit_root: [0; 32],
+        bridge_exits: Vec::new(),
+        imported_bridge_exits: Vec::new(),
+        signature: Signature {
+            r: U256::ZERO,
+            s: U256::ZERO,
+            odd_y_parity: false,
+        },
+    };
+    let certificate_id = certificate.hash();
+    let proof = Proof::new_for_test();
+
+    orchestrator
+        .pending_store
+        .insert_pending_certificate(1.into(), 1, &certificate2)
+        .unwrap();
+
+    orchestrator
+        .pending_store
+        .insert_generated_proof(&certificate_id, &proof)
+        .unwrap();
+
+    orchestrator
+        .state_store
+        .insert_certificate_header(&certificate)
+        .unwrap();
+
+    assert!(orchestrator
+        .state_store
+        .get_certificate_header_by_cursor(1.into(), 1)
+        .unwrap()
+        .is_none());
+
+    let state = LocalNetworkState::default();
+    let result = orchestrator.handle_certifier_result(Ok(crate::CertifierOutput {
+        certificate,
+        height: 0,
+        new_state: state,
```

Comment:
I think need some rebase on main, given that `new_state` is a `LocalNetworkStateData` now in `main`